pool.go

```diff
@@ -366,17 +366,23 @@ func (c *conn) send(e Email) (bool, error) {
 	if err != nil {
 		return true, err
 	}
-	defer w.Close()
 
 	// Get raw message payload.
 	msg, err := e.Bytes()
 	if err != nil {
 		return false, err
```

Comment:
If this triggers, the connection remains unclosed and leaks as the `defer` is gone.